lib/src/main/java/org/esbtools/eventhandler/Promise.java

```diff
@@ -24,22 +24,24 @@
  * An asynchronous capture of potentially yet-to-come responses.
  *
  * <p>Allows code to build a {@link Future} object which will complete at some point in the future
- * with results from the {@link ResponsesHandler} passed to {@link #then(ResponsesHandler)}.
+ * with results from the {@link PromiseHandler} passed to {@link #then(PromiseHandler)}.
  *
  * @see Requester
  *
  * @param <T> The type of requests
- * @param <U> The type of responses
+ * @param <T> The type of responses
  */
-public interface ResponsePromise<T, U> {
+public interface Promise<T> extends Future<T> {
```

Comment:
> I suppose I'm talking less about this interface, and more about how this interface is used in the overall solution. So, we can discuss this under LazyPromise if you'd like; specifically the implementation of get.

This is a fair conversation, but has little to do with the `Promise` interface as you point out correctly. In fact, I think the laziness is the main interesting part of this implementation that is worth chatting about as far as its pros and cons.